ArtOfIllusion/src/artofillusion/RotateViewTool.java

```diff
@@ -208,13 +214,13 @@ private void dragRotateSpace(WidgetMouseEvent e, ViewerCanvas view)
 		dx = dragPoint.x-clickPoint.x;
 		dy = dragPoint.y-clickPoint.y;
 
-		if (controlDown && !e.isShiftDown())
+		if (mouseButtonTwo(e) && controlDown)
```

Comment:
```suggestion
		if (controlDown && !e.isShiftDown())
```
MB2 is already down, no need to redetect.

The !e.isShiftDown() is necessary to to separate this from the next one: If shift is down then both of the blocks with controlDown will try to respond and one of the loses.

What happens is, that in Space and Fly modes you can not temporarily switch between rotate and pan if you remove the !isShift...

This is a bit heretic structure from coding point of view: It is not the common `if(condition){something}, else{something else}` but a `if(condition){something}, if(not condition){something else}` which does not automatically exclude the other when one happens. At the time I ended up with this structure, because I started to have hard time following the logic inside all the nested if-else blocks, so I made each combination their own case. The structure could now be turned more compact.

Reply:
This one works as intended.